lib/beaker/hypervisor/docker.rb

```diff
@@ -68,33 +68,31 @@ def provision
           'Image' => image_name,
           'Hostname' => host.name,
         }
+        container = find_container(host)
 
-        unless host['docker_container_name'].nil?
-          @logger.debug("Looking for an existing container called #{host['docker_container_name']}")
-          existing_container = ::Docker::Container.all.select do |container| container.info['Names'].include? "/#{host['docker_container_name']}" end
-
-          # Prepare to use the existing container or else create it
-          if existing_container.any?
-            container = existing_container[0]
-          else
-            container_opts['name'] = host['docker_container_name']
+        # If the specified container exists, then use it rather creating a new one
+        if container.nil?
+          unless host['mount_folders'].nil?
+            container_opts['HostConfig'] ||= {}
+            container_opts['HostConfig']['Binds'] = host['mount_folders'].values.map do |mount|
+              a = [ File.expand_path(mount['host_path']), mount['container_path'] ]
+              a << mount['opts'] if mount.has_key?('opts')
+              a.join(':')
+            end
           end
-        end
 
-        unless host['mount_folders'].nil?
-          container_opts['HostConfig'] ||= {}
-          container_opts['HostConfig']['Binds'] = host['mount_folders'].values.map do |mount|
-            a = [ File.expand_path(mount['host_path']), mount['container_path'] ]
-            a << mount['opts'] if mount.has_key?('opts')
-            a.join(':')
+          if @options[:provision]
+            if host['docker_container_name']
+              container_opts['name'] = host['docker_container_name']
+            end
+
+            @logger.debug("Creating container from image #{image_name}")
+            container = ::Docker::Container.create(container_opts)
           end
         end
 
-        # If the specified container exists, then use it rather creating a new one
-        if container.nil?
-          @logger.debug("Creating container from image #{image_name}")
-          container = ::Docker::Container.create(container_opts)
-        end
+        raise RuntimeError, "Unable to create container because provision=false" if container.nil?
```

Comment:
error message could be a bit clearer. Mention something that the container couldn't be found ?
Perhaps fail when you do the find part?